community/index/src/main/java/org/neo4j/index/internal/gbptree/GBPTree.java

```diff
@@ -1046,14 +1046,35 @@ void printTree() throws IOException
      * @param printState whether or not to print the tree state.
      * @throws IOException on I/O error.
      */
-    public void printTree( boolean printValues, boolean printPosition, boolean printState ) throws IOException
+    void printTree( boolean printValues, boolean printPosition, boolean printState ) throws IOException
     {
         try ( PageCursor cursor = openRootCursor( PagedFile.PF_SHARED_READ_LOCK ) )
         {
             new TreePrinter<>( bTreeNode, layout, stableGeneration( generation ), unstableGeneration( generation ) )
                 .printTree( cursor, cursor, System.out, printValues, printPosition, printState );
         }
     }
+    // Utility method
+    /**
+     * Print node with given id to System.out, if node with id exists.
+     * @param id the page id of node to print
+     */
+    void printNode( int id ) throws IOException
+    {
+        if ( id < freeList.lastId() )
+        {
+            // Use write lock to avoid adversary interference
```

Comment:
So grabbing a write cursor because some tests have adversarial cursors? The things that adversarial cursors do also happen in real life. Imho instead either change the comment or do this with a read cursor.

Reply:
The thinking was that `printNode`, just as `printTree` should never be used in production but is just a utility method for testing. If you still think we should use read cursor here anyway, I'm ok with that.

Reply:
Well OK, tree printing is a special little thing, sure. I'm OK with that.